### ISSUE TITLE
Unnecessary "ANN101 Missing type annotation for `self` in method"

### ISSUE THREAD

**Issue by artemiogr97** [astral-sh/ruff#10268]
Given the following code contained in my_file.py:

```
class MyClass:
    def do_something(self, my_float) -> int:
        return int(my_float)
```

running "ruff check --select ANN ./my_file.py" gives the following output:

test_ann101.py:2:22: ANN101 Missing type annotation for `self` in method
Found 1 error.

This is useless in most cases, in fact ruff extension for vscode does not show this error.

**Comment by trag1c** [comment]
ANN101 has been deprecated, perhaps you're using a different version of Ruff in your command line?

**Comment by dhruvmanila** [comment]
Hey, thanks for opening this issue. As mentioned by @trag1c the rule has been [deprecated](https://docs.astral.sh/ruff/rules/missing-type-self/) and will be removed in a future release. 

> in fact ruff extension for vscode does not show this error.

It must be that you're running a different Ruff version via the command-line and in VS Code. In case you need to use an old Ruff version, you can also [ignore the rule](https://docs.astral.sh/ruff/settings/#lint_ignore) with:

```toml
[tool.ruff.lint]
ignore = ["ANN101"]
```
